### PR TITLE
`InputText` - add atomic component

### LINKED ISSUE #564: `InputText` - atomic component
#### What would you like to be added:
Atomic component for InputText

- [ ] Must have styling for inline validation
- [ ] Placeholder color (as it's broken on iOS)
- [ ] Styling for error of forms, red outline
- [ ] Update usage across all screens
- [ ] Clear button
- [ ] Ability to accept right side components. Like QR code or Token Symbol
- [ ] Ability to accept titles

image

### PR DESCRIPTION
**Pull request by kyleleow**
#### What kind of PR is this?:

/kind feature
/kind refactor

#### What this PR does / why we need it:
Updated text input in screens below:
- send screen
- convert screen
- add liquidity screen

With the following features:
- ability to accept title
- ability to display on focus outline
- ability to display invalid outline
- ability to clear input
- ability to display inline validation text
- ability to cater for text and numeric input
- ability to accept children prop to display as right-hand-side element

#### Which issue(s) does this PR fixes?:

Fixes #564

#### Additional comments?:
- DEX-related screen are not implemented due to unfinalized design

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [x] Your UI implementation visually matched the rendered design*
- [x] Unit tests*
- [x] Added e2e tests*
- [x] Added translations*

### CODE REVIEW

File: app/components/WalletTextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    inputType,
+    title,
+    valid,
+    inlineValidationText,
+    displayClearButton = false,
+    onClearButtonPress,
+    children,
+    style,
+    ...otherProps
+} = props
```
Comment (thedoublejay) [inline app/components/WalletTextInput.tsx:31]: Can you check your formatter haha

File: app/components/WalletTextInput.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+  const isInvalid = (): boolean => {
+    return valid === false
```
Comment (thedoublejay) [inline app/components/WalletTextInput.tsx:34]: Can we just inline `valid`?

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: app/components/WalletTextInput.tsx (hunk 2 above)
Comment (kyleleow) [reply app/components/WalletTextInput.tsx:36]: Can I keep this for readability purpose? This is due to this function is used in nested ternary operator in theme styling. 

Use case:
``light={tailwind(`bg-white ${isInvalid() ? 'border-error-500' : (isFocus ? 'border-primary-300' : 'border-gray-300')}`)}``

File: app/components/WalletTextInput.tsx (hunk 1 above)
Comment (kyleleow) [reply app/components/WalletTextInput.tsx]: updated

File: app/components/WalletTextInput.tsx (hunk 2 above)
Comment (thedoublejay) [reply app/components/WalletTextInput.tsx:36]: But it can be `!valid` right?

Comment (kyleleow) [reply app/components/WalletTextInput.tsx]: updated

**Review by thedoublejay — APPROVED** [review]

**Comment by izzycsy** [comment]
Reference from #564 

`Acceptance Criteria`

Expected: `Input text` UI to be consistent throughout

Text input in:
- [ ] Send screen
- [ ] Convert screen
- [ ] Add liquidity screen

**Android:**
- [ ] Must have styling for inline validation
- [ ] Placeholder `color` 
- [ ] Styling for error of forms, `red` outline
- [ ] Update usage across all screens
- [ ] `Clear` button
- [ ] Ability to accept right side components. Like QR code or Token Symbol
- [ ] Ability to accept titles

**iOS:**
- [ ] Must have styling for inline validation
- [ ] Placeholder `color` 
- [ ] Styling for error of forms, `red` outline
- [ ] Update usage across all screens
- [ ] `Clear` button
- [ ] Ability to accept right side components. Like QR code or Token Symbol
- [ ] Ability to accept titles

**Pull request merged**